prow/spyglass/gcsartifact_fetcher.go

```diff
@@ -79,12 +79,12 @@ func fieldsForJob(src *gcsJobSource) logrus.Fields {
 func newGCSJobSource(src string) (*gcsJobSource, error) {
 	gcsURL, err := url.Parse(fmt.Sprintf("gs://%s", src))
 	if err != nil {
-		return &gcsJobSource{}, ErrCannotParseSource
+		return &gcsJobSource{}, err
```

Comment:
Are there downstream consumers of this? It's an exported method so perhaps if we want to expose why we can't parse the source, we want a wrapper:


```go
type ErrCannotParseSource struct {
    inner error
}

func (e ErrCannotParseSource) Error() string {
    return fmt.Sprintf("could not create job source from provided source: %v", e.inner)
}

func IsErrCannotParseSource(err error) bool {
    if err == nil {
        return false
    }

   _, isParseErr := err.(*ErrCannotParseSource)
   return isParseErr
}
```